spec/jobs/record_payment_job_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+require "geckoboard"
+
+RSpec.describe RecordPaymentJob do
+  let(:claim) { build(:claim) }
+  let(:payment) { build(:payment, :with_figures, updated_at: DateTime.now, claim: claim) }
+
+  subject { described_class.new }
+
+  it "sends the claim reference, policy and payment date to claims.submitted.ENV dataset" do
+    ClimateControl.modify ENVIRONMENT_NAME: "environment_name" do
+      claim_data = {
+        reference: claim.reference,
+        policy: claim.policy.to_s,
+        performed_at: payment.updated_at.strftime("%Y-%m-%dT%H:%M:%S%:z"),
+      }
+
+      stub_geckoboard_dataset_find_or_create("claims.paid.environment_name")
```

Comment:
`stub_geckoboard_dataset_post` already calls `stub_geckoboard_dataset_find_or_create`, so this call isn't necessary. (There are a few other places like this in the codebase where this call can be removed.)